### PR TITLE
Add config file for AppveyorCI

### PR DESCRIPTION
**Pull request by kpp**
See https://github.com/sodiumoxide/sodiumoxide/issues/223

### CODE REVIEW

**Comment by kpp** [comment]
I don't know why travis failed. But it is ok for appveyor  to fail (hence https://github.com/sodiumoxide/sodiumoxide/issues/223).

**Comment by Dylan-DPC** [comment]
Also you need to add it to the [bors](https://github.com/sodiumoxide/sodiumoxide/blob/master/bors.toml) configuration.

**Comment by kpp** [comment]
I missed that you added a new variable: SODIUM_INC_DIR. Now things look a little bit better:

```
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_wstdlib.h:276:55: error: expected ')', err: true
fatal error: too many errors emitted, stopping now [-ferror-limit=], err: true
thread 'main' panicked at 'Unable to generate bindings: ()', libcore\result.rs:945:5
stack backtrace:
...
```

File: .appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Thanks to https://github.com/starkat99/appveyor-rust
```
Comment (Dylan-DPC) [inline .appveyor.yml]: you need to make an entry in the [bors](https://github.com/sodiumoxide/sodiumoxide/blob/master/bors.toml)  configuration.

**Comment by kpp** [comment]
how about this for travis?

```

matrix:
  allow_failures:
  - rust: nightly
```

**Comment by Dylan-DPC** [comment]
@kpp  
> how about this for travis?

I feel that would be misleading. I would want it to fail on nightly and block on that so we know the build is compatible with the next release. Also in future, we might have features that work only on nightly and thus need the nightly version of the toolkit.

**Comment by Dylan-DPC** [comment]
bors: r+

**Comment by Dylan-DPC** [comment]
bors: r+

**Comment by Dylan-DPC** [comment]
bors: try

**Comment by Dylan-DPC** [comment]
@kpp some issues with [appveyor build](https://ci.appveyor.com/project/Dylan-DPC/sodiumoxide/build/1.0.4). Can you fix them?

**Comment by kpp** [comment]
> Can you fix them?

```
thread 'main' panicked at 'Unable to generate bindings: ()', libcore\result.rs:945:5
```

Also there are many errors like

```
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:304:51: error: unknown type name 'uintptr_t'
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:305:67: error: unknown type name 'uintptr_t'
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:309:8: error: unknown type name 'uintptr_t'
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:317:1: error: a parameter list without types is only allowed in a function definition
```

I don't have a computer with Windows onboard to try to build your project, but in seems like your code is not xplatform right now. So your question is more addressed to your code not to Appveyour build.

Sorry.

**Comment by Dylan-DPC** [comment]
@kpp Ah that's fine. Yah even i'm not on windows so will wait for another co-maintainer/collaborator to check this.

**Comment by kinggoesgaming** [comment]
The build script is failing. most likely trying to find `libsodium`. does this PR only add support for the Appveyor config?

@kpp 

----

```
Running `C:\projects\sodiumoxide\target\debug\build\libsodium-sys-878953a9d9b7de1a\build-script-build`
error: failed to run custom build command for `libsodium-sys v0.1.0 (file:///C:/projects/sodiumoxide/libsodium-sys)`
process didn't exit successfully: `C:\projects\sodiumoxide\target\debug\build\libsodium-sys-878953a9d9b7de1a\build-script-build` (exit code: 101)
--- stdout
cargo:rerun-if-env-changed=SODIUM_LIB_DIR
cargo:rerun-if-env-changed=SODIUM_INC_DIR
cargo:rerun-if-env-changed=SODIUM_STATIC
cargo:rustc-link-search=native=C:\projects\sodiumoxide\deps\libsodium-win64\lib
cargo:rustc-link-lib=static=sodium
--- stderr
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vadefs.h:28:33: error: expected ';' after top level declarator
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:81:1: error: expected identifier
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:81:20: error: expected ';' after top level declarator
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:194:29: error: expected ';' after top level declarator
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:195:30: error: typedef redefinition with different types ('__int64' (aka 'unsigned int') vs 'long long')
C:\Program Files\LLVM\lib\clang\6.0.0\include\stddef.h:51:26: note: previous definition is here
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:304:51: error: unknown type name 'uintptr_t'
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:305:67: error: unknown type name 'uintptr_t'
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:309:8: error: unknown type name 'uintptr_t'
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:317:1: error: a parameter list without types is only allowed in a function definition
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:317:18: error: expected ';' after top level declarator
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt.h:12:1: error: expected identifier
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt.h:12:20: error: expected ';' after top level declarator
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt.h:1976:1: error: a parameter list without types is only allowed in a function definition
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_malloc.h:13:1: error: expected function body after function declarator
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_malloc.h:228:1: error: a parameter list without types is only allowed in a function definition
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_search.h:16:1: error: expected function body after function declarator
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_search.h:206:1: error: a parameter list without types is only allowed in a function definition
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_wstdlib.h:13:1: error: expected function body after function declarator
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_wstdlib.h:276:55: error: expected ')'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_wstdlib.h:275:40: note: to match this '('
fatal error: too many errors emitted, stopping now [-ferror-limit=]
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vadefs.h:28:33: error: expected ';' after top level declarator, err: true
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:81:1: error: expected identifier, err: true
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:81:20: error: expected ';' after top level declarator, err: true
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:194:29: error: expected ';' after top level declarator, err: true
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:195:30: error: typedef redefinition with different types ('__int64' (aka 'unsigned int') vs 'long long'), err: true
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:304:51: error: unknown type name 'uintptr_t', err: true
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:305:67: error: unknown type name 'uintptr_t', err: true
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:309:8: error: unknown type name 'uintptr_t', err: true
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:317:1: error: a parameter list without types is only allowed in a function definition, err: true
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vcruntime.h:317:18: error: expected ';' after top level declarator, err: true
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt.h:12:1: error: expected identifier, err: true
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt.h:12:20: error: expected ';' after top level declarator, err: true
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt.h:1976:1: error: a parameter list without types is only allowed in a function definition, err: true
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_malloc.h:13:1: error: expected function body after function declarator, err: true
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_malloc.h:228:1: error: a parameter list without types is only allowed in a function definition, err: true
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_search.h:16:1: error: expected function body after function declarator, err: true
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_search.h:206:1: error: a parameter list without types is only allowed in a function definition, err: true
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_wstdlib.h:13:1: error: expected function body after function declarator, err: true
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\corecrt_wstdlib.h:276:55: error: expected ')', err: true
.
.
.
Command exited with code 101
```

**Comment by kpp** [comment]
> most likely trying to find libsodium

no

> does this PR only add support for the Appveyor config?

yes

**Comment by kpp** [comment]
So how is it going? We cannot upgrade to sodiumoxide v1.0.0 until you fix this issue.

**Comment by kpp** [comment]
Closed in favor of #269

**Pull request closed without merging**